reana_server/status.py

```diff
@@ -251,11 +262,95 @@ def get_status(self):
         }
 
 
+class QuotaUsageStatus(REANAStatus):
+    """Class to retrieve statistics related to the current REANA users quota usage."""
+
+    def __init__(self, from_=None, until=None, user=None):
+        """Initialise QuotaUsageStatus class.
+
+        :param from_: From which moment in time to collect information. Not
+            implemented yet.
+        :param until: Until which moment in time to collect information. Not
+            implemented yet.
+        :param user: A REANA-DB user model.
+        :type from_: datetime
+        :type until: datetime
+        :type user: reana_db.models.User
+        """
+        super().__init__(from_=from_, until=until, user=user)
+
+    def get_top_five_resource_usage_users(
+        self, resource_filter, order, user_resource_filter=True
+    ):
+        user_resource_subq = (
```

Comment:
After testing locally doing the following change in `helm/configurations/values-dev.yaml` in reanahub/reana
```diff
+notifications:
+  enabled: true
+  system_status: "*/3 * * * *"
+  email_config:
+    receiver: admins@reana.cern.ch
+    sender: notifications@reana.cern.ch
```
and creating a few users and running workflows.

I see:
![image](https://user-images.githubusercontent.com/9035606/97302957-09794d80-185a-11eb-8f1f-12693bd6653a.png)

I've spotted a couple of things:
- Only 3 users are shown instead of five
- The aggregation is not completely correct, because UserResource might have several resources for a user of a different ResourceType, so we need to group by the resource type and do a `sum` (which corresponds currently to [this bit](https://github.com/reanahub/reana-db/blob/c03aa75fed2b0c5d3aa4cc3555bf72ac3a3cf6b0/reana_db/models.py#L156-L166)).

Maybe the best to find a solution for these to details is to write some unit tests, what do you think? It would be definitely faster than trying with the cluster :).

Reply:
Good idea! I can definitely write some unit tests. I had completely missed that one user can have several resources of the same type. By the way, why is this?

Reply:
This is a great question. We've chosen to have a lot of flexibility so we can have [Resources](https://github.com/reanahub/reana-db/blob/a7b7753066e57b93720cd9cbb4db0137a68c4f88/reana_db/models.py#L756) like "meyrin cpu", "general cpu", "xxx cpu" and all of them of [ResourceType](https://github.com/reanahub/reana-db/blob/a7b7753066e57b93720cd9cbb4db0137a68c4f88/reana_db/models.py#L697) `cpu`. So when we want to know how much CPU usage a given user has we need to go through all the kinds.

We have discussed with @mvidalgarcia about this flexibility making things quite complicated in general. Maybe we can simplify the problem by asking ourselves the following two questions:
- Do we really need this flexibility CC @tiborsimko 
- Do we have the simplest design to achieve it (if we really need it)

Reply:
I see, thank you for explaining!

Reply:
@diegodelemos with your suggested changes to `helm/configurations/values-dev.yaml` do I need to rebuild the cluster or can I apply the values to an existing cluster while keeping the existing users?

Reply:
Addressing your comments and my fixes:

> Only 3 users are shown instead of five

I increased the limit from 5 to 10 and now it works as expected. I think it is due to `self.get_top_five_resource_usage_users()` returning an array of tuples, each tuple containing two objects. It seems like `limit()` is not computed over each item in the array but includes the items in the tuples.

> The aggregation is not completely correct, because UserResource might have several resources for a user of a different ResourceType, so we need to group by the resource type and do a sum (which corresponds currently to this bit).

By adding `group_by()` I was able to utilise the `func.sum()` method.

Reply:
> @diegodelemos with your suggested changes to helm/configurations/values-dev.yaml do I need to rebuild the cluster or can I apply the values to an existing cluster while keeping the existing users?

You just need to redeploy and everything should work as expected:
```console
$ reana-dev cluster-undeploy
$ kubectl get pods -w  # wait for all pods to disappear
$ reana-dev cluster-deploy --exclude-components=r-a-vomsproxy,r-a-krb5 --admin-email admin@cern.ch --admin-password 123456  --mode debug
```

Regarding the amendments, I will shortly retest your changes, thanks for updating!